src/lexer.jl

```diff
@@ -34,23 +32,29 @@ mutable struct Lexer{IO_t <: IO}
 
     last_token::Tokens.Kind
     charstore::IOBuffer
+    current_char::Char
+    doread::Bool
 end
 
-Lexer(io) = Lexer(io, position(io), 1, 1, -1, position(io), 1, 1, position(io), Tokens.ERROR, IOBuffer())
-Lexer(str::AbstractString) = Lexer(IOBuffer(str))
+Lexer(io::IO_t, T) where {IO_t} = Lexer{IO_t,T}(io, position(io), 1, 1, position(io), 1, 1, position(io), Tokens.ERROR, IOBuffer(), ' ', false)
```

Comment:
Could you change this to default to `T = Token` and also make sure that only `T <: AbstractToken` are allowed. I noticed this from OhMyREPL failing on this branch.

Reply:
I realize the `Lexer` is not strictly part of public API but then I don't have to change OhMyREPL and perhaps someone is creating `Lexers` with one argument...